inference-engine/src/transformations/src/transformations/common_optimizations/hswish_fusion.cpp

```diff
@@ -262,3 +262,29 @@ ngraph::pass::HSwishFusionWithClampDiv::HSwishFusionWithClampDiv() {
     auto m = std::make_shared<ngraph::pattern::Matcher>(mul, matcher_name);
     register_matcher(m, callback);
 }
+
+NGRAPH_RTTI_DEFINITION(ngraph::pass::HSwishFusionWithHSigmoidMul, "HSwishFusionWithHSigmoidMul", 0);
+
+ngraph::pass::HSwishFusionWithHSigmoidMul::HSwishFusionWithHSigmoidMul() {
+    MATCHER_SCOPE(HSwishFusionWithHSigmoidMul);
+    // Replaces a sub-graph x * HSigmoid(x) with a HSwish op.
+    auto input = ngraph::pattern::any_input();
+    auto hsigmoid_pattern = ngraph::pattern::wrap_type<ngraph::opset6::HSigmoid>({input}, ngraph::pattern::consumers_count(1));
+    auto mul_pattern = ngraph::pattern::wrap_type<ngraph::opset6::Multiply>({input, hsigmoid_pattern});
+
+    ngraph::matcher_pass_callback callback = [=](ngraph::pattern::Matcher &m) {
+        auto mul = std::dynamic_pointer_cast<opset6::Multiply>(m.get_match_root());
+        if (!mul)
+            return false;
+        auto& pattern_to_output = m.get_pattern_value_map();
+        auto hsigmoid = pattern_to_output.at(hsigmoid_pattern).get_node_shared_ptr();
+        auto hswish = std::make_shared<ngraph::opset6::HSwish>(hsigmoid->input_value(0));
+        hswish->set_friendly_name(mul->get_friendly_name());
+        ngraph::copy_runtime_info({hsigmoid, mul}, hswish);
+        ngraph::replace_node(mul, hswish);
+        return true;
```

Comment:
You don't need to cast nodes until you don't use any special methods from operation classes.
Also all this code can be simplified to this:
```suggestion
        const auto& pattern_to_output = m.get_pattern_value_map();
        auto hsigmoid = pattern_to_output.at(hsigmoid_pattern).get_node_shared_ptr();
        auto mul = pattern_to_output.at(mul_pattern).get_node_shared_ptr();

        auto hswish = std::make_shared<ngraph::opset6::HSwish>(pattern_to_output.at(input));
        hswish->set_friendly_name(mul->get_friendly_name());
        ngraph::copy_runtime_info({hsigmoid, mul}, hswish);
        ngraph::replace_node(mul, hswish);
        return true;
```